<!-- Standard Code -->

<html><head>
    
<script type="text/javascript" src="../../ext/adapter/ext/ext-base.js"></script>
<script type="text/javascript" src="../../ext/ext-all-debug.js"></script>
<script type="text/javascript" src="../external/jsunit/app/jsUnitCore.js"></script>
<script type="text/javascript" src="../external/jsmock.js"></script>
<script type="text/javascript" src="helpers.js"></script>

<!-- End Standard Code -->

<script type="text/javascript" src="../ext_overrides.js"></script>

</head><body><script type="text/javascript">

function testDateRenderer() {
  // I added exception handling
  var fn = Ext.util.Format.dateRenderer('D M j y');

  var value = fn('bob');
  assertEquals('on exception', 'Invalid', value);

  value = fn('05/02/08');
  assertEquals('valid date', 'Sat May 2 08', value);
}

function testGetKlass() {
  var mc = new MockControl();
  var store = new Ext.data.Store({
    klass: 'bob',
    reader: new Ext.data.JsonReader({},
                [ 'id', 'klass', 'text', 'estimated', 'quicktip' ])
  });

  var record = new store.recordType({});
  assertUndefined('getKlass un-init', record.getKlass());
  record.join(store);
  assertEquals('getKlass inited', store.klass, record.getKlass());
}

</script></body></html>
